lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1595,6 +1611,43 @@ bool QuakeBridgeVisitor::VisitCallExpr(clang::CallExpr *x) {
                                          reportNegateError, isAdjoint,
                                          isControl, /*paramCount=*/3);
 
+    // See if this is a custom unitary.
+    std::string maybeUnitaryGenerator = funcName.str() + "_generator_";
+    // Extract number of targets
+    size_t targetCount = 0;
+    std::string opName;
+    for (auto name : customOperationNames) {
+      opName = name.str();
+      if (name.consume_front(maybeUnitaryGenerator)) {
+        targetCount = std::stoi(name.str());
+        break;
+      }
+    }
+    if (targetCount > 0) {
+      // Extract number of parameters
+      std::size_t paramCount = [&]() {
+        std::size_t count = 0;
+        for (auto arg : args) {
+          auto argTy = arg.getType();
+          if (isa<FloatType>(argTy))
+            count++;
+          else if (auto ptrTy = dyn_cast<cc::PointerType>(argTy)) {
+            if (isa<FloatType>(ptrTy.getElementType()))
+              count++;
+          }
+        }
+        return count;
+      }();
+      buildOp<quake::CustomUnitarySymbolOp>(
+          builder, loc, args, negations, reportNegateError, isAdjoint,
+          isControl, paramCount, targetCount, true);
+      auto &customUnitary = builder.getBlock()->back();
+      auto srefAttr =
+          SymbolRefAttr::get(StringAttr::get(builder.getContext(), opName));
+      customUnitary.setAttr("generator", srefAttr);
```

Comment:
@schweitzpgi Are you cool with this (note it was my idea)? This was the only way I could figure out how to add a `SymbolRefAttr` to this operation, and inherit from `QuakeOperator` in the tablegen so we get all the semantics of a quake op.

Reply:
Ideally we'd have a distinct argument in the `inp` list for it. (Adding a layer of safety, since it can't be omitted or lost should the op get cloned for example.)  I think we can do that, but I have to do a little experimenting.

Reply:
Patches sent.

Reply:
Addressed in 3dba0f518f7e5e9c62fc8296e31e867152dd0753